core/transport/src/main/proto/proto/journal/journal.proto

```diff
@@ -16,9 +16,20 @@ import "proto/journal/table.proto";
 // and Spark. We use protobuf version 2.5.0 instead, which is compatible with Hadoop and Spark.
 //
 
-// next available id: 52
+/**
+ * Unique operation id.
+ */
+message JournalOpPId {
+    optional int64 mostSignificantBits = 1;
+    optional int64 leastSignificantBits = 2;
+}
+
+// next available id: 53
 message JournalEntry {
+  // shared fields.
   optional int64 sequence_number = 1;
+  optional JournalOpPId operationId = 52;
```

Comment:
Is it intended that the order here is messed up? LOL

Reply:
No idea!